### PR TITLE
UTILS: Remove (or Suggest) under-utilized dependencies

### PR DESCRIPTION
**Pull request by ashiklom**
## Description

More and more of us are getting frustrated by how many dependencies PEcAn has. This package starts to chip away at the dependency list of `PEcAn.utils` by removing some dependencies outright (e.g. `plyr`) and making others Suggests. For the latter case, I also added a helper function, `need_packages`, that can be used to throw an error if `Suggest`-ed packages are not installed.

## Motivation and Context

Dependencies are the worst.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [X] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My change requires a change to the documentation.
- [X] I have updated the CHANGELOG.md.
- [X] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by ashiklom** [comment]
Tagging @infotroph and @serbinsh as reviewers because I know how much they hate dependencies 😉 , and @dlebauer because a lot of this code is stuff he wrote originally.

File: base/utils/DESCRIPTION [hunk 1]
```diff
@@ -23,38 +23,36 @@ Description: The Predictive Ecosystem Carbon Analyzer
...
+    raster,
+    rjags,
+    sp,
+    testthat (>= 2.0.0),
+    xtable
 License: FreeBSD + file LICENSE
 Copyright: Authors
 LazyData: true
-Require: hdf5, plyr
+Require: hdf5
```
Comment (infotroph) [inline base/utils/DESCRIPTION]: I think R ignores this field. Should this be SystemRequirements instead?

File: base/utils/DESCRIPTION [hunk 2]
```diff
@@ -23,38 +23,36 @@ Description: The Predictive Ecosystem Carbon Analyzer
...
-    sp,
     stringi,
     udunits2 (>= 0.11),
-    xtable,
     XML
 Suggests:
+    coda (>= 0.18),
     MASS,
+    data.table,
+    ggplot2,
```
Comment (infotroph) [inline base/utils/DESCRIPTION]: Should be in alphabetical order

File: base/utils/R/need_packages.R [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  have <- vapply(pkgs, requireNamespace, logical(1), quietly = TRUE)
+  if (any(!have)) {
+    msg <- sprintf(
+      "The following packages are required but not installed: %s",
+      paste0("`", pkgs, "`", collapse = ", ")
+    )
+    PEcAn.logger::logger.severe(msg)
+  }
+  invisible(pkgs)
+}
```
Comment (infotroph) [inline base/utils/R/need_packages.R]: 👍👍

File: base/utils/DESCRIPTION (hunk 1 above)
Comment (ashiklom) [reply base/utils/DESCRIPTION]: I'm just gonna drop it altogether. All of the system requirements are for dependent packages, so it's probably their responsibility to make sure this is documented.

File: base/utils/DESCRIPTION (hunk 2 above)
Comment (ashiklom) [reply base/utils/DESCRIPTION]: Good catch. Addressed in 7a87488aff42f720bcf7deef05d54fe03c93f07f.

**Pull request merged**